pages/_app.js

```diff
@@ -14,14 +14,110 @@ export default function App({ Component, pageProps }) {
     setProjects(projects.filter((project) => project.id !== id));
   }
 
+  const complexityOrder = { Beginner: 0, Intermediate: 1, Advanced: 2 };
+  function handleSortProjectsByComplexityStartHigh() {
+    setProjects(
+      projects.toSorted((a, b) => {
+        return complexityOrder[b.complexity] - complexityOrder[a.complexity];
+      })
+    );
+  }
+  function handleSortProjectsByComplexityStartLow() {
+    setProjects(
+      projects.toSorted((a, b) => {
+        return complexityOrder[a.complexity] - complexityOrder[b.complexity];
+      })
+    );
+  }
+
+  function durationToHours(duration) {
+    const durationValue = parseInt(duration);
+    if (duration.toLowerCase() && duration.includes("hour")) {
+      return durationValue;
+    } else if (duration.toLowerCase() && duration.includes("day")) {
+      return durationValue * 24;
+    } else if (duration.toLowerCase() && duration.includes("week")) {
+      return durationValue * 24 * 7;
+    } else if (duration.toLowerCase() && duration.includes("month")) {
+      return durationValue * 24 * 30;
+    } else if (duration.toLowerCase() && duration.includes("year")) {
+      return durationValue * 24 * 365;
+    }
+    return duration;
+  }
+  function handleSortProjectsByDurationStartLong() {
+    setProjects(
+      projects.toSorted((a, b) => {
+        const durationA = durationToHours(a.duration);
+        const durationB = durationToHours(b.duration);
+        return durationB - durationA;
+      })
+    );
+  }
+  function handleSortProjectsByDurationStartShort() {
+    setProjects(
+      projects.toSorted((a, b) => {
+        const durationA = durationToHours(a.duration);
+        const durationB = durationToHours(b.duration);
+        return durationA - durationB;
+      })
+    );
+  }
```

Comment:
First of all, great job on getting this to work. That is the most important thing, and sorting isnt simple.

If you are looking to clean things up a bit we could combine these 2 functions into 1. 

i.e. something like

``` js
function sortProjectsByDuration(direction = 'long') {
  setProjects(
    projects.toSorted((a, b) => {
      const durationA = durationToHours(a.duration);
      const durationB = durationToHours(b.duration);
      return direction === 'long' ? durationB - durationA : durationA - durationB;
    })
  );
}

```

That means when we call the function we would need to pass an argument. 
i.e.

``` html
<StyledSortButton type="button" onClick={() => onSortDurationStart("short")}>
  <input type="radio" id="duration-short" name="sort" value="duration" />
  <label htmlFor="duration-short">Duration: short to long</label>
</StyledSortButton>

<StyledSortButton type="button" onClick={() =>onSortDurationStart("long")}>
  <input type="radio" id="duration-long" name="sort" value="duration" />
  <label htmlFor="duration-long">Duration: long to short</label>
</StyledSortButton>
```

You would also need to keep in mind that the prop names would need to change.

Reply:
Hey Peter, thank you. I understand what you did there and did the same. It's only working for one of the cases (long to short) but not the other (short to long). And I can not figure out why.

Reply:
It requires a fair bit of refactoring so feel free to skip it this time. 

But something like this seems to work...

ModalSort.js

``` js
export default function ModalSort({
  onSortComplexityStartHigh,
  onSortComplexityStartLow,
  onSortDurationStart,
  onToogleSortModal,
}) {
  return (
    <ModalWrapper>
      <p>Sort projects by ...</p>
      <StyledSortButton
        type="button"
        onClick={() => onSortDurationStart("short")}
      >
        <input type="radio" id="duration-short" name="sort" value="duration" />
        <label htmlFor="duration-short">Duration: short to long</label>
      </StyledSortButton>

      <StyledSortButton
        type="button"
        onClick={() => onSortDurationStart("long")}
      >
        <input type="radio" id="duration-long" name="sort" value="duration" />
        <label htmlFor="duration-long">Duration: long to short</label>
      </StyledSortButton>
```

index.js

``` js
export default function HomePage({
  projects,
  onFilterProjects,
  onResetFilters,
  onSortProjectsByComplexityStartHigh,
  onSortProjectsByComplexityStartLow,
  onSortProjectsByDuration,
}) {
  const [modalSort, setModalSort] = useState(false);
  const [modalFilter, setModalFilter] = useState(false);

  function toogleSortModal() {
    setModalSort(!modalSort);
  }

  function handleSortComplexityStartHigh() {
    onSortProjectsByComplexityStartHigh();
    toogleSortModal();
  }
  function handleSortComplexityStartLow() {
    onSortProjectsByComplexityStartLow();
    toogleSortModal();
  }
  function handleSortDurationStart(direction) {
    onSortProjectsByDuration(direction);
    toogleSortModal();
  }

  function toogleFilterModal() {
    setModalFilter(!modalFilter);
  }

  return (
    <>
      <StyledHeadline>DIY APP</StyledHeadline>
      <Navigation />

      <button type="button" onClick={toogleSortModal}>
        sort projects by ...
      </button>
      {modalSort && (
        <ModalSort
          onToogleSortModal={toogleSortModal}
          onSortComplexityStartHigh={handleSortComplexityStartHigh}
          onSortComplexityStartLow={handleSortComplexityStartLow}
          onSortDurationStart={handleSortDurationStart}
        />
      )}
```

_app.js

``` js

 function sortProjectsByDuration(direction = "long") {
    setProjects(
      projects.toSorted((a, b) => {
        const durationA = durationToHours(a.duration);
        const durationB = durationToHours(b.duration);
        return direction === "long"
          ? durationB - durationA
          : durationA - durationB;
      })
    );
  }
  // ....

return (
    <>
      <GlobalStyle />
      <Component
        {...pageProps}
        projects={displayedProjects}
        onAddProject={handleAddProject}
        onDeleteProject={handleDeleteProject}
        onFilterProjects={handleProjectFilter}
        onResetFilters={resetProjectFilter}
        onSortProjectsByComplexityStartHigh={
          handleSortProjectsByComplexityStartHigh
        }
        onSortProjectsByComplexityStartLow={
          handleSortProjectsByComplexityStartLow
        }
        onSortProjectsByDuration={sortProjectsByDuration}
      />
    </>
  );

```

Reply:
Thanks! That helped me find my mistake (I forgot to give the argument "direction" in the index.js).